plugin/file/setup.go

```diff
@@ -121,6 +121,7 @@ func fileParse(c *caddy.Controller) (Zones, error) {
 				return Zones{}, c.Errf("unknown property '%s'", c.Val())
 			}
 		}
+		reader.Close()
```

Comment:
Would be better to do `defer reader.Close()` right after opening.

Reply:
Oh, sorry,  the reader is closed in github.com/miekg/dns subNext() .
```
func (zp *ZoneParser) subNext() (RR, bool) {
	if rr, ok := zp.sub.Next(); ok {
		return rr, true
	}

	if zp.sub.osFile != nil {	//this 
		zp.sub.osFile.Close()
		zp.sub.osFile = nil
	}

	if zp.sub.Err() != nil {
		// We have errors to surface.
		return nil, false
	}

	zp.sub = nil
	return zp.Next()
}
```
Thanks.